pytest_httpserver/blocking_http_server.py

```diff
@@ -0,0 +1,163 @@
+from queue import Empty
+from queue import Queue
+from typing import Any
+from typing import Mapping
+from typing import Optional
+from typing import Pattern
+from typing import Union
+
+from werkzeug.wrappers import Request
+from werkzeug.wrappers import Response
+
+from pytest_httpserver.httpserver import METHOD_ALL
+from pytest_httpserver.httpserver import UNDEFINED
+from pytest_httpserver.httpserver import HeaderValueMatcher
+from pytest_httpserver.httpserver import HTTPServer
+from pytest_httpserver.httpserver import QueryMatcher
+from pytest_httpserver.httpserver import RequestHandler
+from pytest_httpserver.httpserver import URIPattern
+
+
+class BlockingRequestHandler(RequestHandler):
+    """
+    Decorates the :py:class:`RequestHandler` to allow responding synchronously.
+
+    This class should only be instantiated inside the implementation of the :py:class:`BlockingHttpServer`.
+    """
+
+    def __init__(self, request: Request, response_queue: Queue):
+        super().__init__(None)
+        self.request = request
+        self.response_queue = response_queue
+
+    def respond_with_data(self, *args, **kwargs):
+        self._respond_via_queue(super().respond_with_data, *args, **kwargs)
+
+    def respond_with_response(self, *args, **kwargs):
+        self._respond_via_queue(super().respond_with_response, *args, **kwargs)
+
+    def _respond_via_queue(self, respond_method, *args, **kwargs):
```

Comment:
I'm not sure what about the other methods such as `respond_with_handler` and similar (defined in `RequestHandler` but not overridden here).

Do they need also be implemented here?

Reply:
It does not make much sense to implement that method, since the response is synchronous.

Reply:
Added respond_with_handler to make it complete.